python/dgl/data/bitcoinotc.py

```diff
@@ -56,14 +95,47 @@ def _load(self, filename):
             rate = data[row_mask][:, 2]
             g.add_edges(edges[:, 0], edges[:, 1])
             g.edata['h'] = rate.reshape(-1, 1)
-            self.graphs.append(g)
+            self._graphs.append(g)
+
+    def has_cache(self):
+        graph_path = os.path.join(self.save_path, 'dgl_graph.bin')
+        return os.path.exists(graph_path)
+
+    def save(self):
+        graph_path = os.path.join(self.save_path, 'dgl_graph.bin')
+        save_graphs(graph_path, self.graphs)
 
-    def __getitem__(self, idx):
-        return self.graphs[idx]
+    def load(self):
+        graph_path = os.path.join(self.save_path, 'dgl_graph.bin')
+        self._graphs = load_graphs(graph_path)[0]
+
+    @property
+    def graphs(self):
+        return self._graphs
 
     def __len__(self):
         return len(self.graphs)
 
+    def __getitem__(self, item):
+        return self.graphs[item]
+
     @property
     def is_temporal(self):
         return True
+
+    def _extract_gz(self, file, target_dir, overwrite=False):
+        if os.path.exists(target_dir) and not overwrite:
+            return
+        print('Extracting file to {}'.format(target_dir))
+        fname = os.path.basename(file)
+        makedirs(target_dir)
+        out_file_path = os.path.join(target_dir, fname[:-3])
```

Comment:
still, please avoid using `fname[:-3]`.

Reply:
At lease you should add some comment here for fnam[:-3]